parade/action_manager.go

```diff
@@ -0,0 +1,156 @@
+package parade
+
+import (
+	"github.com/google/uuid"
+	"github.com/treeverse/lakefs/logging"
+	"sync"
+	"time"
+)
+
+const (
+	defaultWorkers     = 5
+	defaultChannelSize = 1000
+	defaultMaxTasks    = 500
+	defaultWaitTime    = time.Millisecond * 300
+	defaultErrWaitTime = time.Millisecond * 300
+	defaultMaxDuration = time.Minute * 30 // Todo(guys): change this
+)
+
+// ManagerProperties defines the configuration properties of an ActionManager
+type ManagerProperties struct {
+	Workers     int            // number of goroutines handling tasks
+	ChannelSize int            // size of the channel containing tasks for workers
+	MaxTasks    int            // max tasks requested in every ownTasks request
+	WaitTime    *time.Duration // time to wait if OwnTasks returned no tasks.
```

Comment:
Not sure why these are pointers and the `int`s are values, identified by being zero.

Reply:
@arielshaqed Because 0 is a valid argument for the pointers. 
And I didn't wan't to override it with a default value.